pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

```diff
@@ -0,0 +1,1827 @@
+						Body: "ewogICAgInNwZWMiOiB7CiAgICAgICJzaWduYXR1cmUiOiB7CiAgICAgIH0KICAgIH0KfQ==",
+					},
+				},
+			},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "Bundle payload body signature content is empty",
+			args: args{
+				bundle: &bundle.RekorBundle{
+					Payload: bundle.RekorPayload{
+						Body: "ewogICAgInNwZWMiOiB7CiAgICAgICAgInNpZ25hdHVyZSI6IHsKICAgICAgICAiY29udGVudCI6ICIiCiAgICAgICAgfQogICAgfQp9",
+					},
+				},
+			},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "Bundle payload body is not a valid JSON",
+			args: args{
+				bundle: &bundle.RekorBundle{
+					Payload: bundle.RekorPayload{
+						Body: "ewogICJzcGVjIjosLCB7CiAgICAic2lnbmF0dXJlIjogewogICAgICAiY29udGVudCI6ICJNRVVDSVFDeWVtOEdjcjBzUEZNUDdmVFhhekNONTdOY041K01qeEp3OU9vMHgyZU0rQUlnZGdCUDk2Qk8xVGUvTmRiakhiVWViMEJVeWU2ZGVSZ1Z0UUV2NU5vNXNtQT0iCiAgICB9CiAgfQp9",
+					},
+				},
+			},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "Bundle payload body signature content is correct",
+			args: args{
+				bundle: &bundle.RekorBundle{
+					Payload: bundle.RekorPayload{
+						Body:           "ewogICJzcGVjIjogewogICAgInNpZ25hdHVyZSI6IHsKICAgICAgImNvbnRlbnQiOiAiTUVVQ0lRQ3llbThHY3Iwc1BGTVA3ZlRYYXpDTjU3TmNONStNanhKdzlPbzB4MmVNK0FJZ2RnQlA5NkJPMVRlL05kYmpIYlVlYjBCVXllNmRlUmdWdFFFdjVObzVzbUE9IgogICAgfQogIH0KfQ==",
+						LogID:          "samplelogID",
+						IntegratedTime: 12345,
+					},
+				},
+			},
+			want:    "MEUCIQCyem8Gcr0sPFMP7fTXazCN57NcN5+MjxJw9Oo0x2eM+AIgdgBP96BO1Te/NdbjHbUeb0BUye6deRgVtQEv5No5smA=",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getBundleSignatureContent(tt.args.bundle)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getBundleSignatureContent() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("getBundleSignatureContent() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_AttestContainerSignatures(t *testing.T) {
+	type fields struct {
+		verifyFunction             func(context context.Context, ref name.Reference, co *cosign.CheckOpts) ([]oci.Signature, bool, error)
+		fetchImageManifestFunction func(ref name.Reference, options ...remote.Option) (*remote.Descriptor, error)
+		skippedImages              map[string]bool
+		rekorURL                   url.URL
+	}
+
+	emptyCheckOptsFunction := func(url.URL) *cosign.CheckOpts {
+		co := &cosign.CheckOpts{}
+		co.RekorClient = new(rekor.Rekor)
+		rootCert, _, _ := GenerateRootCa()
+		rootPool := x509.NewCertPool()
+		rootPool.AddCert(rootCert)
+		co.RootCerts = rootPool
+
+		return co
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		status  corev1.ContainerStatus
+		want    []string
+		wantErr bool
+	}{
+		{
+			name: "Attest image with signature",
+			fields: fields{
+				verifyFunction: func(context context.Context, ref name.Reference, co *cosign.CheckOpts) ([]oci.Signature, bool, error) {
```

Comment:
can you verify received parameters? we need to know what we are sending to cosign libraries